.github/workflows/codeql.yaml

```
@@ -0,0 +1,30 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [development/*, stabilization/*, hotfix/*]
+  pull_request:
+    branches: [development/*, stabilization/*, hotfix/*]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        language: [python, typescript, go, javascript]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+
+      - name: Build and analyze
+        uses: github/codeql-action/analyze@v2
```

When setting up the workflow via the GitHub ui, the following config get's added on the matrix build:

```suggestion
      - name: Perform CodeQL Analysis
        uses: github/codeql-action/analyze@v2
        with:
          category: "/language:${{ matrix.language }}"
 ```
 
 Not sure if it's required, but it seems like a logical configuration to me.